src/BuildScriptGenerator/Python/PythonPlatform.cs

```diff
@@ -67,26 +68,31 @@ public LanguageDetectorResult Detect(ISourceRepo sourceRepo)
 
         public BuildScriptSnippet GenerateBashBuildScriptSnippet(BuildScriptGeneratorContext context)
         {
-            var buildProperties = new Dictionary<string, string>();
-            buildProperties[ManifestFilePropertyKeys.OperationId] = context.OperationId;
+            var manifestFileProperties = new Dictionary<string, string>();
 
             var packageDir = GetPackageDirectory(context);
             var virtualEnvName = GetVirtualEnvironmentName(context);
-            if (string.IsNullOrEmpty(packageDir))
+
+            if (!string.IsNullOrWhiteSpace(packageDir) && !string.IsNullOrWhiteSpace(virtualEnvName))
```

Comment:
if neither is provided, we default to virtual env with its default name

Reply:
Right, it does not change that(line 83). I just rearranged to code to make it more readable.